### DISCUSSION TITLE
websocket: limit maximum message length

### DISCUSSION THREAD

**Discussion by iot49** [miguelgrinberg/microdot#192] [Q&A]
Is there a way to limit the maximum size microdot websocket.receive() accepts?

Motivation: excessive message length can result in MemoryError, possibly crashing the application. Malicious agents could do exploit this. And even non-malicious code could accidentally send an excessively long message.

**Comment by miguelgrinberg** [comment]
This is in my list already. There is currently no limitation, but I do intend to add this for safety.

**Reply by beyonlo** [reply]
Actually I'm using the websocket on Microdot accepting JSON messages with 8KB - that is my biggest message to receive. But I always was thinking that too. Would be nice if exists a limit for security and MemoryError, so I can to limite my max limit to receive 9KB for example. In many tests when I was receiving messages bigger than 10KB cause slow process on the Microcontroller. So, +1 for this feature.

**Reply by miguelgrinberg** [reply]
Just added this to the main branch. By default the limit set in `Request.max_body_length` is now also a limit for incoming WebSocket messages. You can use a different size by setting `WebSocket.max_message_length`. Set to 0 to disable size checks on WebSocket messages.

**Reply by beyonlo** [reply]
@miguelgrinberg Excellent. I still using the old Microdot as async of course. I see that you released in the last week the Microfot v2 where it do not support anymore threads, just asyncio as core. Congratulations for this new release, I think that Microdot it is now smaller (and faster?) since that do not need to support the threads anymore, right? Anyway, in near future I will read your [Migration Guide](https://microdot.readthedocs.io/en/stable/migrating.html) to migrate to v2, so I will to test this new feature `WebSocket.max_message_length` too :)

Thank you for your excellent work in this project!
